File: generated-code/example/go/model/structure.go
package model

import "io"
import . "trans_gen_test/stream"
import "fmt"

type Structure struct {
    Text string
    FloatNumber float32
    DoubleNumber float64
}

func NewStructure(text string, floatNumber float32, doubleNumber float64) Structure {
    return Structure {
        Text: text,
        FloatNumber: floatNumber,
        DoubleNumber: doubleNumber,
    }
}

func ReadStructure(reader io.Reader) Structure {
    var text string
    text = ReadString(reader)
    var floatNumber float32
    floatNumber = ReadFloat32(reader)
    var doubleNumber float64
    doubleNumber = ReadFloat64(reader)
    return Structure {
        Text: text,
        FloatNumber: floatNumber,
        DoubleNumber: doubleNumber,
    }
}

func (structure Structure) Write(writer io.Writer) {
    text := structure.Text
    WriteString(writer, text)
    floatNumber := structure.FloatNumber
    WriteFloat32(writer, floatNumber)
    doubleNumber := structure.DoubleNumber
    WriteFloat64(writer, doubleNumber)
}

func (structure Structure) String() string {
    stringResult := "{ "
    stringResult += "Text: "
    text := structure.Text
    stringResult += "\"" + text + "\""
    stringResult += ", "
    stringResult += "FloatNumber: "
    floatNumber := structure.FloatNumber
    stringResult += fmt.Sprint(floatNumber)
    stringResult += ", "
    stringResult += "DoubleNumber: "
    doubleNumber := structure.DoubleNumber
    stringResult += fmt.Sprint(doubleNumber)
    stringResult += " }"
    return stringResult
}

File: generated-code/example/typescript/src/model/structure.ts
import { StreamWrapper } from "../stream-wrapper";

export class Structure {
    text: string
    floatNumber: number
    doubleNumber: number

    constructor(text: string, floatNumber: number, doubleNumber: number) {
        this.text = text;
        this.floatNumber = floatNumber;
        this.doubleNumber = doubleNumber;
    }

    static async readFrom(stream: StreamWrapper): Promise<Structure> {
        let text;
        text = await stream.readString();
        let floatNumber;
        floatNumber = await stream.readFloat();
        let doubleNumber;
        doubleNumber = await stream.readDouble();
        return new Structure(text, floatNumber, doubleNumber)
    }

    async writeTo(stream: StreamWrapper) {
        let text = this.text;
        await stream.writeString(text);
        let floatNumber = this.floatNumber;
        await stream.writeFloat(floatNumber);
        let doubleNumber = this.doubleNumber;
        await stream.writeDouble(doubleNumber);
    }
}